### DISCUSSION TITLE
Is there Symfony CLI in symfony-docker?

### DISCUSSION THREAD

**Discussion by linaran** [dunglas/symfony-docker#288] [Q&A] [answered]
Note:
I only recently began checking out Symfony, my main framework is Django so this question might be missing something obvious.
My goal is to get familiar with symfony and keep all of its dependencies inside docker containers (I connect with VsStudioCode to write code).

**Is `Symfony CLI` available in `Symfony-docker` (at least dev-container), and if not, why?**

My attempt to find it:
```
docker-compose exec php sh
symfony  # not found

or

docker-compose exec php symfony
```
Neither worked, `symfony` as a command was missing. From what I can tell, I don't necessarily need `Symfony CLI`, I can get by with `composer` but `Symfony CLI` seems like a nice quality of life improvement. The docs and tutorials rely on it, I imagine symfony users use it regularly and I assume it's something similar to `manage.py` in `Django`. It might not be useful for starting web servers as those start automatically through the container but it's a nice to have for other things?

Thanks for the answers 🙏

**Comment by dunglas** [comment, marked as answer]
Hi,

No it isn't installed because it is useless when using this project.

What you basically need is to run `docker compose exec php bin/console` instead of `symfony console` and `docker compose exec php composer` instead of `symfony composer`.

All other commands (running a local web server etc) shouldn't be needed when using this project, as everything is already provided as Docker containers.

**Reply by maximebarber** [reply]
Thanks ! 

What about commands like:
- symfony server:ca:install
- symfony book:check-requirements
- symfony new --version=6.2-1 --book guestbook

What is the logic for executing those commands? I got them from https://symfony.com/doc/6.2/the-fast-track/en/2-project.html

**Reply by maxhelias** [reply]
The `symfony server:ca:install` command, is not needed as this project provide Caddy, which automatically create a certificate. 

Note : Using SymfonyCLI or this project is 2 different ways of providing a local web server. And this project is not intended to provide the elements of the book, but a basis for a production, development and CI ready (https://github.com/dunglas/symfony-docker#features).
